docs/api-reference/expression-estimator.md

```diff
@@ -0,0 +1,181 @@
+```
+c : with(f = c * 9 / 5 + 32, cold = f < 60, hot = f > 90 ; -float(cold) + float(hot))
+```
+As demonstrated above, the with-expression is useful when an expression value is needed multiple times in a larger expression. It is also useful when dealing with complicated or significant constants:
+```
+    ticks => with(
+        ticksPerSecond = 10000000L,
+        ticksPerHour = ticksPerSecond \* 3600,
+        ticksPerDay = ticksPerHour \* 24,
+        day = ticks / ticksPerDay,
+        dayEpoch = 1 ;
+        (day + dayEpoch) % 7)
+```
+This computes the day of the week from the number of ticks (as an Int64) since the standard .Net DateTime epoch (01/01/0001
+in the idealized Gregorian calendar). Assignments are used for number of ticks in a second, number of ticks in an hour,
+number of ticks in a year, and the day of the week for the epoch. For this example, we want to map Sunday to zero, so,
+since the epoch is a Monday, we set dayEpoch to 1. If the epoch were changed or we wanted to map a different day of the week to zero,
+we'd simply change dayEpoch. Note that ticksPerSecond is defined as 10000000L, to make it an Int64 value (8 byte integer).
+Without the L suffix, ticksPerDay will overflow Int32's range.
+
+## Functions
+
+The expression transform supports many useful functions.
+
+General unary functions that can accept an operand of any type are listed in the following table.
+
+| **​Name** | ​ **Meaning** | ​ **Comments** |
+| --- | --- | --- |
+| ​isna | ​test for na | Returns a boolean value indicating whether the operand is an NA value. |
+| ​na | the na value | ​Returns the NA value of the same type as the operand (either float or double). Note that this does not evaluate the operand, it only uses the operand to determine the type of NA to return, and that determination happens at compile time. |
+| ​default | ​the default value | ​​Returns the default value of the same type as the operand. For example, to map NA values to default values, use x ?? default(x). Note that this does not evaluate the operand, it only uses the operand to determine the type of default value to return, and that determination happens at compile time. For numeric types, the default is zero. For boolean, the default is false. For text, the default is empty. |
+
+The unary conversion functions are listed in the following table. An NA operand produces an NA, or throws if the type does not support it.
+A conversion that doesn't succeed, or overflow also result in NA or an exception. The most common case of this is when converting from text,
+which uses the standard conversion parsing. When converting from a floating point value (float or double) to an integer value
+(Int32 or Int64), the conversion does a truncate operation (round toward zero).
+
+| **​Name** | ​ **Meaning** | ​ **Comments** |
+| --- | --- | --- |
+| ​bool | ​convert to BL | The operand must be text or boolean. |
```

Comment:
>BL [](start = 22, length = 2)

we don't have these types in our public API I think.

Reply:
Resolved in follow-up PR: https://github.com/dotnet/machinelearning/pull/4614/files#r362319131

#Resolved